src/Assets/AssetDataRegistry.php

```diff
@@ -60,8 +60,8 @@ public function __construct( Api $asset_api ) {
 	 */
 	protected function init() {
 		add_action( 'init', array( $this, 'register_data_script' ) );
+		add_action( 'wp_print_scripts', array( $this, 'enqueue_asset_data' ), 1 );
 		add_action( 'wp_print_footer_scripts', array( $this, 'enqueue_asset_data' ), 1 );
-		add_action( 'admin_print_footer_scripts', array( $this, 'enqueue_asset_data' ), 1 );
```

Comment:
Is there possibility for double printed data with this change? I see that you've added a `! wp_script_is( $this->handle, 'done' )` later in `enqueue_asset_data` but if that was intended to prevent double printed data, that will only work in the admin right?  

Regarding this change impacting the admin, there are at least 2 Woo extensions (but likely more) that I know of that are relying on data exposed on `wcSettings`. This change will need tested with those:

- WC-Admin.
- AutomateWoo.

Due to the impact, it might be wise to do this in it's own PR so the change has more visibility.

Reply:
I'm happy to log this and let someone else work on it. I just needed to be unblocked to get MailPoet working locally.

Reply:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5052